### PR TITLE
Feature Flag for Legacy P2P Support

### PR DESCRIPTION
**Pull request by erikgrinaker**
This wires up the node to use either the new or old P2P stack based on the `node.useLegacyP2P` constant, using the new stack by default. It's a real mess, but it passes the simplest E2E tests:

```
test/e2e $ make && ./build/runner -f networks/simple.toml 
I[2021-02-05|12:46:44.401] Removing Docker containers and networks      
I[2021-02-05|12:46:44.625] Generating testnet files in "networks/simple" 
I[2021-02-05|12:46:44.633] Starting initial network nodes...            
I[2021-02-05|12:46:44.633] Starting transaction load (16 workers)...    
I[2021-02-05|12:46:46.580] Node validator01 up on http://127.0.0.1:5701 
I[2021-02-05|12:46:48.464] Node validator02 up on http://127.0.0.1:5702 
I[2021-02-05|12:46:50.374] Node validator03 up on http://127.0.0.1:5703 
I[2021-02-05|12:46:52.285] Node validator04 up on http://127.0.0.1:5704 
I[2021-02-05|12:46:52.285] Waiting for initial height 1...              
I[2021-02-05|12:46:52.310] Waiting for all nodes to reach height 7...   
I[2021-02-05|12:46:59.038] Ending transaction load after 112 txs (7.8 tx/s)... 
I[2021-02-05|12:46:59.045] Waiting for all nodes to reach height 12...  
I[2021-02-05|12:47:05.777] Running tests in ./tests/...                 
ok  	github.com/tendermint/tendermint/test/e2e/tests	1.787s
```

Branched off of #5969.

### CODE REVIEW

**Comment by alexanderbez** [comment]
@erikgrinaker at any point did this PR work with regards to e2e tests? I keep seeing `chain stalled at unknown height` regardless if the legacy stack is used or not.

**Comment by erikgrinaker** [comment]
It works with the simple.toml network, i.e. the simplest possible 4-node network. The ci.toml network won't work, e.g. because it requires support for seed mode. This PR is just a basic proof of concept, there's a lot of work remaining before it's at parity with the legacy stack.

**Comment by alexanderbez** [comment]
I would like to mark this PR r4r by having support for the feature flag only and having it build & pass. Did you envision anything specifically for this PR?

**Comment by erikgrinaker** [comment]
This PR was just a quick-and-dirty proof of concept that I threw together in a couple of hours on my last day to see if the full stack was functional. It probably has bugs and omissions, and needs to be cleaned up. Whether you want to take this PR across the line or start fresh with a new PR is your call, but it wasn't really intended to be merged in its current form.

**Comment by alexanderbez** [comment]
Ok, thanks for the clarification. Let me figure out if I can get seed mode to work as well in this PR and get CI to pass and then we can go from there in future PRs.

**Comment by erikgrinaker** [comment]
Getting seed mode to work with the new stack isn't trivial, we were discussing writing a separate daemon for it. As for why the legacy stack doesn't work, I'm not sure -- might have just forgotten some wiring or gotten something wrong somewhere.

**Comment by alexanderbez** [comment]
I see. So I'm a bit lost on how we can even finish this PR then, unless we change CI to just run `simple` network for now.

**Comment by erikgrinaker** [comment]
You could default to the legacy stack, fix the issues causing it to break here, and clean up the code -- then implement the remaining P2P work until all E2E testnets pass. Or alternatively implement all of the outstanding P2P work first, before wiring up the node.

**Comment by alexanderbez** [comment]
Let's leave the default to true to use the legacy p2p stack. Note, there are no issues with the legacy stack in this PR -- false alarm. I'll clean this up and merge it shortly.

File: node/node.go [hunk 1]
```diff
@@ -488,6 +559,52 @@ func createTransport(
 	)
 }
 
+func createPeerManager(config *cfg.Config, p2pLogger log.Logger, nodeID p2p.NodeID) (*p2p.PeerManager, error) {
+	options := p2p.PeerManagerOptions{
+		MaxConnected:           64,
```
Comment (melekes) [inline node/node.go:564]: will this be configurable in the future?

**Review by melekes — APPROVED** [review]

File: node/node.go (hunk 1 above)
Comment (alexanderbez) [reply node/node.go:564]: Yes, these will come from configuration :)

Comment (alexanderbez) [reply node/node.go:564]: These are mainly just placeholders I believe to get things running in e2e tests.

**Pull request merged**